src/js/components/CheckBox/CheckBox.js

```diff
@@ -66,6 +66,11 @@ const CheckBox = forwardRef(
       initialValue: defaultChecked,
     });
 
+    if (disabled) {
+      /* eslint-disable no-param-reassign */
+      focusProp = false; 
+    }
+
     const [focus, setFocus] = useState(focusProp);
     useEffect(() => setFocus(focusProp), [focusProp]);
```

Comment:
```suggestion
    const [focus, setFocus] = useState(focusProp);
    useEffect(() => {
      // don't allow checkbox to have focus when disabled
      if (disabled && focusProp) setFocus(false);
      else setFocus(focusProp);
    }, [disabled, focusProp]);
```
I'm thinking we should do something like this. Then we don't have to re-assign a function parameter. Also useEffect should be used for code that happens during lifecycle of the component, such as removing focus when disabled switches from false to true.

Reply:
Yes sure. Using useEffect we won't be re-assigning function param. 
Thanks for the suggestion. I've made the changes as suggested.